src/components/NewCommentForm.tsx

```diff
@@ -1,9 +1,139 @@
+  const [currentEmail, setCurrentEmail] = useState('');
+  const [currentBody, setCurrentBody] = useState('');
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [isNameError, setIsNameError] = useState(false);
+  const [isEmailError, setIsEmailError] = useState(false);
+  const [isBodyError, setIsBodyError] = useState(false);
+  const [currentError, setCurrentError] = useState<Errors | null>(null);
+
+  const resetForm = (reason: 'submit' | 'clear') => {
+    if (reason === 'clear') {
+      setCurrentName('');
+      setCurrentEmail('');
+    }
+
+    setCurrentBody('');
+    setIsNameError(false);
+    setIsEmailError(false);
+    setIsBodyError(false);
+  };
+
+  const handleNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentName(event.target.value);
+    if (isNameError) {
+      setIsNameError(false);
+    }
+  };
+
+  const handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setCurrentEmail(event.target.value);
+    if (isEmailError) {
+      setIsEmailError(false);
+    }
+  };
+
+  const handleBodyChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setCurrentBody(event.target.value);
+    if (isBodyError) {
+      setIsBodyError(false);
+    }
+  };
+
+  const validateName = (testedName: string) => {
+    if (testedName.trim() === '') {
+      setIsNameError(true);
+    }
+
+    return testedName.trim().length > 0;
+  };
+
+  const validateEmail = (testedEmail: string) => {
+    if (!/\S+@\S+\.\S+/.test(testedEmail)) {
+      setIsEmailError(true);
+    }
+
+    return /\S+@\S+\.\S+/.test(testedEmail);
+  };
+
+  const validateCommentBody = (testedBody: string) => {
+    if (testedBody.trim() === '') {
+      setIsBodyError(true);
+    }
+
+    return testedBody.trim().length > 0;
+  };
+
+  const validateForm = () => {
+    const nameValid = validateName(currentName);
+    const emailValid = validateEmail(currentEmail);
+    const bodyValid = validateCommentBody(currentBody);
+
+    return nameValid && emailValid && bodyValid;
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (validateForm()) {
+      setIsAddingComment(true);
+    }
+  };
+
+  useEffect(() => {
+    if (!isAddingComment) {
+      return;
+    }
+
+    setCurrentError(null);
+
+    newComment({
+      name: currentName,
+      email: currentEmail,
+      body: currentBody,
+      postId: postId,
+    })
+      .then((data: unknown) => {
+        const dataAsComment = data as Comment;
+
+        if (!dataAsComment || !dataAsComment.id) {
+          setCurrentError(Errors.CommentCreating);
+          setIsAddingComment(false);
+
+          return;
+        }
+
+        setComments(prevComments => [...prevComments, dataAsComment]);
+        resetForm('submit');
+        setIsAddingComment(false);
+      })
+      .catch(() => {
+        setCurrentError(Errors.CommentCreating);
+        setIsAddingComment(false);
```

Comment:
Ensure that the user receives clear feedback when a comment creation fails. Consider adding more descriptive error messages or handling different error states more distinctly.